src/UhdmAst.cpp

```diff
@@ -169,18 +169,18 @@ bool is_expr_context(vpiHandle obj_h) {
         case vpiAssignStmt:
         case vpiAssignment:
         case vpiContAssign:
+        case vpiFuncCall:
         case vpiReturn: {
             return true;
         }
         case vpiBegin:
-        case vpiFuncCall:
         case vpiTaskCall:
         case vpiCaseItem: {
             return false;
         }
         default: {
             UINFO(3, "Encountered unhandled parent type in " << __FUNCTION__ << std::endl);
-            return false;
+            return true;
         }
```

Comment:
We shouldn't really rely on the default case, can you add proper types instead?

Reply:
Done. I added throwing an error in the default case, because it can result in error in later phase of verilation which can be harder to find.